### DISCUSSION TITLE
Adding reverse edges best practice?

### DISCUSSION THREAD

**Discussion by Jeriousman** [pyg-team/pytorch_geometric#4087] [Q&A]
I understood from @[rusty1s](https://github.com/rusty1s) that adding reverse edges could improve embeddings update in general as the reverse relations are also covered. But should it be done only when src and dst nodes are in the same node types even in the heterogeneous graph? Or even if src and dst nodes are in different node types, we better add reverse edges? What is the best practice in general and what are the trailing reasons? Thank you in advance.

**Comment by rusty1s** [comment]
The reason for adding reverse edges is to guarantee a message passing flow in both directions. For example, given user and product types and a "user->buys->product" relation, messages would get only send to product nodes, and you would lose out on crucial information if you do not include the reverse relation. As a general rule of thumb, you should always include reverse edges.

In general, this also holds true for directed homogeneous graphs, where you often see that the graph is converted to an undirected one first before inputted into a neural network.
